File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.java
public class Codec {

    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        StringBuilder encodedString = new StringBuilder();
        for(String str: strs){
            encodedString.append(str.length()).append("#").append(str);
        }
        return encodedString.toString();
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        List<String> decodedList = new ArrayList<>();
        int i = 0;
        while(i < s.length()){
            int j = i;
            while(s.charAt(j) != '#') j++;
            
            int length = Integer.valueOf(s.substring(i, j));
            i = j + 1 + length;
            decodedList.add(s.substring(j + 1, i));
        }
        return decodedList;
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(strs));

File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        #we assume that only upto 256 ascii characters are going to be present
        #so we can take the 257th ascii character as a separator
        sep = chr(257)
        return sep.join(strs) 

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        sep = chr(257)
        return s.split(sep)


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))